mapproxy/test/system/test_dimensions.py

```diff
@@ -0,0 +1,246 @@
+# This file is part of the MapProxy project.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import division
+import functools
+import os 
+import pytest
+from mapproxy.test.system import SysTest
+from mapproxy.util.ext.wmsparse.util import parse_datetime_range,parse_duration
+from mapproxy.util.ext.wmsparse.duration import parse_datetime
+
+from mapproxy.request.wms import (
+    WMS100MapRequest,
+    WMS111MapRequest,
+    WMS130MapRequest,
+    WMS111CapabilitiesRequest,
+    WMS130CapabilitiesRequest,
+    WMS100CapabilitiesRequest,
+    WMS110MapRequest,
+    WMS110FeatureInfoRequest,
+    WMS110CapabilitiesRequest,
+)
+
+expected_dim = {'time': {
+                        'values': '2020-09-22T11:20:00Z,2020-09-22T13:20:00Z,2020-09-22T15:20:00Z', 
+                        'default': '2020-09-22T14:20:00Z'}, 
+                'dim_reference_time': {
+                        'values': '2020-09-22T11:20:00Z,2020-09-22T13:20:00Z,2020-09-22T15:20:00Z', 
+                        'default': '2020-09-22T14:20:00Z'}}
+
+@pytest.fixture(scope="module")
+def config_file():
+    return "dimension.yaml"
+
+class TestDimensions(SysTest):
+
+    def setup(self):
+        self.common_req = WMS130MapRequest(
+            url="/service?", param=dict(service="WMS", version="1.3.0")
+        )
+
+        self.common_req2 = WMS111MapRequest(
+            url="/service?", param=dict(service="WMS", version="1.1.1")
+        )
+
+        self.common_req3 = WMS110MapRequest(
+            url="/service?", param=dict(service="WMS", version="1.1.0")
+        )
+
+        self.common_req4 = WMS100MapRequest(
+            url="/service?", param=dict(service="WMS", version="1.0.0")
+        )
+
+
+    def test_parsing_datetime(self):
```

Comment:
This, test_parse_duration and test_parse_datetime_range would benefit from pytest.mark.parametrize (https://github.com/mapproxy/mapproxy/blob/master/mapproxy/test/unit/test_srs.py#L103..L135). More tests with less code.

Also: Tests can be functions, no need for test methods. And these tests should be moved to unit/test_times.py